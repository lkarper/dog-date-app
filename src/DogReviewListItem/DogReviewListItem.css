.DogReviewListItem__li {
    list-style: none;
    padding: 10px;
    margin: 20px auto;
    width: 100%;
}

.DogReviewListItem__h3 {
    text-align: center;
}

.DogReviewListItem__star-outer-container {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
}

@media screen and (min-width: 750px) {
    .DogReviewListItem__star-outer-container {
        justify-content: start;
    }
}