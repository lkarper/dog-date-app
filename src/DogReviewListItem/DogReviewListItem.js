import React, { useContext } from 'react';
import moment from 'moment';
import UserContext from '../contexts/UserContext';
import { Link } from 'react-router-dom';

const DogReviewListItem = (props) => {

    const context = useContext(UserContext);

    const { review } = props;
    const { 
        id,
        reviewer,
        date_created,
        friendliness_dogs,
        friendliness_people,
        playing_interest,
        obedience,
        profile_accuracy,
        location_suitability,
    } = review;

    const dog = context.allDogs.find(dog => dog.id === review.dog_id);

    const averageRating = (
        friendliness_dogs +
        friendliness_people +
        playing_interest +
        obedience +
        profile_accuracy + 
        location_suitability 
    ) / 6;

    return (
        <li>
            <h3><Link to={`/reviews/${id}`}>Review of {dog.name}</Link></h3>
            <p>Reviewed by {reviewer} on {moment(date_created).format("MMMM Do YYYY, h:mm a")}</p>
            <p>Overall rating: {averageRating.toFixed(2)} stars</p>
        </li>
    )
}

export default DogReviewListItem;