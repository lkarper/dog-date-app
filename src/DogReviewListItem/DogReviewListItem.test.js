import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import DogReviewListItem from './DogReviewListItem';

describe('DogReviewListItem component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<DogReviewListItem />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<DogReviewListItem />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
