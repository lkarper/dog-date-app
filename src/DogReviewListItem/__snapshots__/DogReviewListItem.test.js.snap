// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogReviewListItem component renders the UI as expected 1`] = `
<DogReviewListItem
  noImage={false}
  review={
    Object {
      "date_created": "",
      "friendliness_dogs": 0,
      "friendliness_people": 0,
      "id": "",
      "location_suitability": 0,
      "obedience": 0,
      "playing_interest": 0,
      "profile_accuracy": 0,
      "review_title": "",
      "reviewer": "",
    }
  }
>
  <li
    className="DogReviewListItem__li error"
  >
    <p>
      Error: could not load review info.  Check your connection and the URL and try again.
    </p>
  </li>
</DogReviewListItem>
`;
